packages/pirania/files/usr/lib/lua/portal/portal.lua

```diff
@@ -57,4 +57,12 @@ function portal.set_page_content(title, main_text, logo, link_title, link_url, b
     return db:insert({portal=data})
 end
 
+function portal.update_captive_portal(daemonized)
+    if (daemonized) then
+        utils.execute_daemonized('captive-portal update')
```

Comment:
I don't think that captive-portal supports concurrent executions of`captive-portal update` so doing it demonized will lead to strange bugs. For example when receiving N vouchers from a peer (because you are joining the network or they have been just created) it will run many instances of captive-portal update demonized, right?

Reply:
Hmm interesting,
For the case of receiving N vouchers from a peer, captive-portal update will be run just once synchronously here at [generate_vouchers](https://github.com/spiccinini/lime-packages/blob/1322c6867c26330e0fd865367e776718d2fb6d38/packages/shared-state-pirania/files/etc/shared-state/hooks/pirania-vouchers/generate_vouchers).
We could end up running captive-portal update concurrently if the node receives two or more rpcd requests for vouchers creations and/or invalidation tough. I think the chances are low for this scenario but we can do something to prevent it.

Reply:
Ah ok!